plugins/PaintTool/PaintTool.py

```diff
@@ -343,9 +391,46 @@ def event(self, event: Event) -> bool:
 
         return False
 
+    def getRequiredExtraRenderingPasses(self) -> list[str]:
+        return ["selection_faces", "picking_selected"]
+
     @staticmethod
     def _updateScene(node: SceneNode = None):
         if node is None:
             node = Selection.getSelectedObject(0)
         if node is not None:
-            Application.getInstance().getController().getScene().sceneChanged.emit(node)
\ No newline at end of file
+            Application.getInstance().getController().getScene().sceneChanged.emit(node)
+
+    def _onSelectionChanged(self):
+        super()._onSelectionChanged()
+
+        self.setActiveView("PaintTool" if len(Selection.getAllSelectedObjects()) == 1 else None)
+        self._updateState()
+
+    def _updateState(self):
```

Comment:
I think it'd be better to add a word or two on _what_ state this is. Either in documentation or through renaming the method.